R/untarDwd.R

```diff
@@ -0,0 +1,117 @@
+################################################################################
+#' Extract files from a tar archive downloaded
+#'
+#' This function untar and unzip dataset previously downloaded from url. Prior to untar, the function can check if untar
+#' was previously performed (untar/unzip files exist locally).
+#'
+#' @param tarfile A character vector containing path to tarfile.
+#' @param dir A character string representing path to output directory.
+#' @param env The environment to search for objects.
+#' @param checkhash A logical argument. If TRUE, check if untar/unzip files are found locally and cross-check their checksum value
+#'        with value logged in dbHash from previous untar/unzip event. When checksums match, a message raises indicating the files
+#'        are already properly untar/unzip. When checksums don't match or untar/unzip files don't exist locally, untar/unzip occurs and
+#'        checksum compiles. If FALSE, untar/unzip occurs even if the files exist locally. Default is FALSE.
+#' @param dbHash A character string. The path to the database file where checksum values are logged If the named
+#'        database does not yet exist, one is created. Default is "dbHash.sqlite".
+#' @param quick A logical argument. If TRUE, checksum is compiled using the combination of the filename and its size.
+#'        If FALSE, cheksum is compiled using the object. Default is FALSE.
+#' @importFrom utils unzip untar
+#'
+#' @export
+#' @docType methods
+#'
+#' @author Melina Houle
+#' @examples
+#' datasetName <-"KNN"
+#' dfile <- c("kNN-LandCover.tar")
+#' algo <- "xxhash64"
+#' outdir <- "C:/temp"
+#' sim <- SpaDES::simInit(times = list(start = 2010.0, end = 2020.0, timeunit = "year"), params = list(),
+#' modules = list(), paths = list(outputPath=outdir))
+#' dwdData(urls, datasetName, dfile = dfile, outdir, sim, module, checkhash = FALSE, cascade = FALSE)
+#' tar <- file.path(outdir, datasetName, dfile)
+#' untarDwd(tar, outdir, checkhash = FALSE, dbHash = "dbHash.sqlite", quick = FALSE)
+untarDwd <-function(tarfile, dir, env= env, checkhash = FALSE, dbHash = "dbHash.sqlite", quick=FALSE){
```

Comment:
why isn't this function used above when downloading data with the `cascade` option? instead, it seems you've duplicated functionality.